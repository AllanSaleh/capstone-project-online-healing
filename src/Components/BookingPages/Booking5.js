import React from 'react';

const Booking5 = () => {
  const choices = [
    {
      id: 1,
      content: 'Depression',
    },
    {
      id: 2,
      content: 'Stress and Anxiety',
    },
    {
      id: 3,
      content: 'Relationship issues',
    },
    {
      id: 4,
      content: 'Family conflicts',
    },
    {
      id: 5,
      content: 'Trauma and Abuse',
    },
    {
      id: 5,
      content: 'Eating disorder',
    },
  ];
  return (
    <div className="h-firstsection px-sides pt-navbar">
      <h1 className="text-title uppercase mb-4">
        letâ€™s match you with the right therapist
      </h1>
      <h3 className="text-subtitle opacity-50 text-justify">
        Please fill out this short questionnaire to provide some general and
        anonymous background about you and the issues you'd like to deal with in
        online therapy. It would help us match you with the most suitable
        therapist for you.
      </h3>

      <div className="flex flex-col justify-evenly h-2/3 max-w-3xl my-16 mx-auto px-8 py-4 shadow-md">
        <div>Are there any issues you'd like to focus on?</div>

        {choices.map((choice) => (
          <div>
            <input
              type="radio"
              id={choice.id}
              name="counselor_options"
              value={choice.content}
            />
            <label className="ml-2" htmlFor={choice.id}>
              {choice.content}
            </label>
          </div>
        ))}

        <div className="flex justify-between">
          <button
            type="button"
            className="uppercase bg-blue-dark h-12 w-32 text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
          >
            back
          </button>
          <button
            type="button"
            className="uppercase bg-blue-dark h-12 w-32 text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
          >
            next
          </button>
        </div>
      </div>
    </div>
  );
};

export default Booking5;
