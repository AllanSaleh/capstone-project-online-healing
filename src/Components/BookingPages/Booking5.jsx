import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';

const Booking5 = () => {
  window.scrollTo(0, 0);
  const choices = [
    {
      id: 1,
      content: 'Depression',
    },
    {
      id: 2,
      content: 'Stress and Anxiety',
    },
    {
      id: 3,
      content: 'Relationship Issues',
    },
    {
      id: 4,
      content: 'Family Conflicts',
    },
    {
      id: 5,
      content: 'Trauma and Abuse',
    },
    {
      id: 6,
      content: 'Eating Disorders',
    },
  ];

  const history = useHistory();
  const [answer, setAnswer] = useState('');
  const handleRadioChange = (event) => {
    setAnswer(event.target.value);
  };
  console.log(answer);
  const PrevPage = () => {
    // Put firestore code here!
    history.push('/Booking4');
  };
  const NextPage = () => {
    // Put firestore code here!
    history.push('/Booking6');
  };

  return (
    <div className="lg:h-firstsection px-sides pt-navbar">
      <h1 className="text-3xl lg:text-title uppercase my-8">
        letâ€™s match you with the right therapist
      </h1>
      <h3 className="text-md lg:text-subtitle opacity-50 text-justify">
        Please fill out this short questionnaire to provide some general and anonymous background
        about you and the issues you'd like to deal with in online therapy. It would help us match
        you with the most suitable therapist for you.
      </h3>

      <div className="flex flex-col justify-evenly w-full md:max-w-md lg:max-w-2xl my-16 mx-auto px-8 py-4 shadow-md">
        <div className="text-md lg:text-subtitle">
          Are there any issue that you would like to focus on?
        </div>

        <div className="h-1/2 flex flex-col justify-around my-8">
          {choices.map((choice) => (
            <div className="flex items-center" key={choice.id}>
              <input
                type="radio"
                id={choice.id}
                name="counselor_options"
                value={choice.content}
                onChange={handleRadioChange}
              />
              <label className="text-md lg:text-paragraph ml-2" htmlFor={choice.id}>
                {choice.content}
              </label>
            </div>
          ))}
        </div>
        <div className="flex justify-between">
          <button
            onClick={() => PrevPage()}
            type="button"
            className="uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
          >
            back
          </button>
          <button
            onClick={() => NextPage()}
            type="button"
            className="uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition-all duration-300"
          >
            next
          </button>
        </div>
      </div>
    </div>
  );
};

export default Booking5;
