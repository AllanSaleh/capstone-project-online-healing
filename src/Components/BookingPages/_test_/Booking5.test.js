import * as React from 'react';
import Booking5 from "../Booking5";
import * as renderer from "react-test-renderer";

it('Booking5 renders correctly', () => {
  const tree = renderer
    .create(<Booking5/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});