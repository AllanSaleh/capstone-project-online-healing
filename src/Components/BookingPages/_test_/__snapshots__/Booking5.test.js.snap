// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking5 renders correctly 1`] = `
<div
  className="h-firstsection px-sides pt-navbar"
>
  <h1
    className="text-3xl lg:text-title uppercase my-8"
  >
    letâ€™s match you with the right therapist
  </h1>
  <h3
    className="text-md lg:text-subtitle opacity-50 text-justify"
  >
    Please fill out this short questionnaire to provide some general and anonymous background about you and the issues you'd like to deal with in online therapy. It would help us match you with the most suitable therapist for you.
  </h3>
  <div
    className="flex flex-col justify-evenly w-full md:max-w-md lg:max-w-2xl my-16 mx-auto px-8 py-4 shadow-md"
  >
    <div
      className="text-md lg:text-subtitle"
    >
      Are there any issue that you would like to focus on?
    </div>
    <div
      className="h-1/2 flex flex-col justify-around my-8"
    >
      <div
        className="flex items-center"
      >
        <input
          id={1}
          name="counselor_options"
          type="radio"
          value="Depression"
        />
        <label
          className="text-md lg:text-paragraph ml-2"
          htmlFor={1}
        >
          Depression
        </label>
      </div>
      <div
        className="flex items-center"
      >
        <input
          id={2}
          name="counselor_options"
          type="radio"
          value="Stress and Anxiety"
        />
        <label
          className="text-md lg:text-paragraph ml-2"
          htmlFor={2}
        >
          Stress and Anxiety
        </label>
      </div>
      <div
        className="flex items-center"
      >
        <input
          id={3}
          name="counselor_options"
          type="radio"
          value="Relationship Issues"
        />
        <label
          className="text-md lg:text-paragraph ml-2"
          htmlFor={3}
        >
          Relationship Issues
        </label>
      </div>
      <div
        className="flex items-center"
      >
        <input
          id={4}
          name="counselor_options"
          type="radio"
          value="Family Conflicts"
        />
        <label
          className="text-md lg:text-paragraph ml-2"
          htmlFor={4}
        >
          Family Conflicts
        </label>
      </div>
      <div
        className="flex items-center"
      >
        <input
          id={5}
          name="counselor_options"
          type="radio"
          value="Trauma and Abuse"
        />
        <label
          className="text-md lg:text-paragraph ml-2"
          htmlFor={5}
        >
          Trauma and Abuse
        </label>
      </div>
      <div
        className="flex items-center"
      >
        <input
          id={6}
          name="counselor_options"
          type="radio"
          value="Eating Disorders"
        />
        <label
          className="text-md lg:text-paragraph ml-2"
          htmlFor={6}
        >
          Eating Disorders
        </label>
      </div>
    </div>
    <div
      className="flex justify-between"
    >
      <button
        className="uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition duration-300"
        onClick={[Function]}
        type="button"
      >
        back
      </button>
      <button
        className="uppercase bg-blue-dark h-12 w-32 text-md lg:text-subtitle rounded-lg border hover:bg-transparent hover:border-blue-dark hover:text-blue-dark transition-all duration-300"
        onClick={[Function]}
        type="button"
      >
        next
      </button>
    </div>
  </div>
</div>
`;
